modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -1240,4 +1278,52 @@ private void cleanUpPartitionsResources(Set<ReplicationGroupId> partitionIds) {
             LOG.error("Unable to clean zones resources", e);
         }
     }
+
+    /**
+     * Lock the zones replica list for any changes. {@link #hasLocalPartition(ZonePartitionId)} must be executed under this lock always.
+     *
+     * @param zoneId Zone id.
+     * @return Stamp, which must be used for further unlock.
+     */
+    public long lockZoneIdForRead(int zoneId) {
```

Comment:
I'd rather rename it a bit to lockZoneForRead and not ZoneId.

Reply:
Fixed